packages/a11y/src/clear.js

```diff
@@ -1,11 +1,11 @@
 /**
  * Clear the a11y-speak-region elements.
  */
-const clear = function() {
-	const regions = document.querySelectorAll( '.a11y-speak-region' );
-	for ( let i = 0; i < regions.length; i++ ) {
-		regions[ i ].textContent = '';
+export default function clear() {
+	const regions = Array.from(
+		document.getElementsByClassName( 'a11y-speak-region' )
+	);
+	for ( const region of regions ) {
+		region.textContent = '';
 	}
```

Comment:
Is `Array.from` necessary? As far as I can tell, `getElementsByClassName` is iterable without needing to be coerced to an array.

Apparently there are some [interesting "gotchas" when using `for ... in`](https://developer.mozilla.org/en-US/docs/Web/API/NodeList#Example), not sure if those apply to `for ... of`.

I guess, generally speaking, I don't consider the previous implementation to be very problematic, at least enough that we'd do this dance of converting to an array just to use `for ... of`.

Reply:
@aduth `getElementsByClassName` returns an `HTMLCollection`, which is not iterable. You may be thinking of a `NodeList`, which _is_ iterable; that's the type returned from other functions like `querySelectorAll`. It's one of those weird inconsistencies caused by different browser APIs being created several years apart.

I went with `getElementsByClassName` because it's faster than `querySelectorAll`.

[`for ... of` is essentially just the no-gotcha version of `for ... in`.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of#Difference_between_for...of_and_for...in) The `for ... of` in JS is equivalent to the `for ... in` or `foreach` in most other languages. It's one of those cases where JavaScript implemented something weirdly early on (e.g. `var`), resulting in a non-weird alternative being added later (e.g. `let`).

I could switch back to `querySelectorAll` and then I wouldn't have to convert to an array before using `for ... of`. Alternatively, I could keep the switch to `getElementsByClassName`, but go back to the old `for` loop, and that also wouldn't require an array conversion. Which do you prefer? Personally, I prefer keeping the `for ... of` loop simply for ease of reading.

Reply:
I don't have a strong preference. It's fine.

But if part of the argument is that `getElementsByClassName` is more performant, I might imagine (not verified) that the cast to an Array type is possibly counteracting any benefit.

Reply:
I tried to switch to `querySelectorAll` and use the `for ... of` loop on that, but the TypeScript linting said that it wasn't iterable. And, because we're targeting ES5, that is correct. There is a TypeScript compiler option called `downlevelIteration` that allows you to use the `for ... of` syntax for ES5 targets by rewriting it to a standard `for` loops upon compilation.

However, that seems like a bit much effort for this one little thing, so I've decided to just revert to using an old-fashioned `for` loop while keeping the switch to `getElementsByClassName`.

Reply:
Actually, looking closer, it looks like we're actually targeting ES6? Argh, maybe I'll investigate this later, but I think this PR is good to go.

Reply:
ES6 should be available in this module, yes. There's a few where we're running code directly, but those are typically not ones we're running the browser.

You can tell based on the `main` of the package's `package.json`. If it's pointing to a file in `build`, then it can be assumed to be safe to use ES6:

https://github.com/WordPress/gutenberg/blob/b4bfbb768618470a81882b8c4b6096603fb7f4ad/packages/a11y/package.json#L21

If there's any issues from the tooling in using ES6 in these packages, that might be something we'd want to look into.

Reply:
To be clear, what I meant by "targeting ES6" was that it appeared that this package was compiling to ES6, so using `for ... of` with `querySelectorAll` would not need to be rewritten to run in the browser.

But looking again, I can't remember what I saw that made me think the package was targeting an ES6 output. Maybe something changed recently, or maybe I just misread something earlier. Oh well. :man_shrugging: